src/Assets/AssetDataRegistry.php

```diff
@@ -66,11 +66,25 @@ public function __construct( Api $asset_api ) {
 	 * Hook into WP asset registration for enqueueing asset data.
 	 */
 	protected function init() {
-		add_action( 'init', array( $this, 'register_data_script' ) );
+		if ( $this->is_site_editor() ) {
+			add_action( 'enqueue_block_editor_assets', array( $this, 'register_data_script' ) );
```

Comment:
I guess it _should_ be okay when restricted to the site editor 🤔

Reply:
Still, I wonder if in a future iteration we should expose a separate API on AssetDataRegistry that is _explicitly_ for blocks to register their asset data rather (and thus enqueued differently) than accounting for multiple locations here. That would help with future maintenance as well if things change.

Reply:
>there may be other Woo Extensions using the AssetDataRegistry which aren't explicitly using it for blocks. Have you double-checked this?

>I guess it should be okay when restricted to the site editor

Yeah, this has only really changed within the context of the site editor. And are we aware of any plugins (of our own or third party) that pass data to the site editor in a non-blocks context? It seems like a bit of an edge case but definitely worth thinking about/a look into.

Reply:
> Still, I wonder if in a followup iteration we should expose a separate API on AssetDataRegistry that is explicitly for blocks to register their asset data rather (and thus enqueued differently) than accounting for multiple locations here

Agree on this! This is just a patch. We will revisit the entire process during the next cooldown!

Reply:
> This is dangerous because there may be other Woo Extensions using the AssetDataRegistry which aren't explicitly using it for blocks. Have you double-checked this?
> I guess it should be okay when restricted to the site editor 🤔

That's a good point: I _think_ so, but IMO it's hard to be 100% sure at this point without any further investigation. We have https://github.com/woocommerce/woocommerce-blocks/issues/9354 as a FU task for a permanent fix.

Reply:
@tjcafferkey @gigitux @nerrad do you think we should prioritize revising this before releasing the patch, just to make sure we won't end up breaking things elsewhere, or are we a-ok with keeping the patch as-is and revisiting later?

Reply:
I would release as-is, but instead of waiting for the cooldown, work on a proper fix now. In this way, we can:
- Apply the new changes if we break something.
- Work on a proper solution without the rush.

Reply:
I tend to be more conservative in such cases so my vote would be for spending some extra time investigating alternatives & possible consequences before releasing, but if most of us are comfortable with releasing as-is I agree with your proposed approach @gigitux